src/agent/rustjail/src/container.rs

```diff
@@ -1437,16 +1437,18 @@ impl LinuxContainer {
             Some(unistd::getuid()),
             Some(unistd::getgid()),
         )
-        .context(format!("cannot change onwer of container {} root", id))?;
+        .context(format!("Cannot change owner of container {} root", id))?;
 
         if config.spec.is_none() {
-            return Err(anyhow!(nix::Error::EINVAL));
+            return Err(anyhow!("Cannot create the container: the spec is none"));
         }
 
         let spec = config.spec.as_ref().unwrap();
 
         if spec.linux.is_none() {
-            return Err(anyhow!(nix::Error::EINVAL));
+            return Err(anyhow!(
+                "Cannot create the container: the linux spec is none"
```

Comment:
Interestingly, both this check and the one above are technically redundant because the first thing the function does is call `validate()` which already performs these checks.